File: resources/js/components/FormTextarea.js
import React from 'react'
import PropTypes from 'prop-types'

class FormTextarea extends React.Component {
  render() {
    return (
      <div className="mb-3">
        <label htmlFor={this.props.name}>{this.props.placeholder}</label>
        <textarea
          style={{ height: this.props.height ?? '100px' }}
          className={
            this.props.field.error ? 'form-control is-invalid' : 'form-control'
          }
          name={this.props.name}
          value={this.props.field.value}
          onChange={this.props.onChange}
          // placeholder={this.props.placeholder}
        ></textarea>
        {this.props.field.error && (
          <label className="invalid-feedback" htmlFor={this.props.name}>
            {this.props.field.error}
          </label>
        )}
      </div>
    )
  }
}

FormTextarea.propTypes = {
  field: PropTypes.shape({
    error: PropTypes.string,
    value: PropTypes.string,
  }),
  name: PropTypes.string,
  height: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
}

export default FormTextarea

File: resources/js/components/FormTextarea.fs
module FormTextarea

open Feliz
open Types

[<ReactComponent>]
let FormTextarea(name: string, 
                 placeholder: string, 
                 height: int option, 
                 field: Field, 
                 onChange: Browser.Types.Event -> unit) : ReactElement =
    Html.div [
        prop.className "mb-3"
        prop.children [
            Html.label [
                prop.htmlFor name
                prop.text placeholder
            ] 
            Html.textarea [
                prop.style [
                    style.height (Option.defaultValue 100 height)
                ]
                prop.className 
                    (if String.length field.error > 0 then "form-control is-invalid" else "form-control") 
                prop.name name
                prop.value field.value
                prop.onChange onChange
            ]
            if String.length field.error > 0 then Html.label [
                prop.className "invalid-feedback"
                prop.htmlFor name
                prop.text field.error
            ]
        ]
    ]